### PR TITLE
Allow array syntax for group.Sync

### PR DESCRIPTION
**Pull request by mikhail**

### CODE REVIEW

File: kingpin/schema.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 
 SCHEMA_1_0 = {
-    'type': 'object',
+    'type': ['object', 'array'],
```
Comment (diranged) [inline kingpin/schema.py]: Lets chat about this. I think that we need an "outer" schema here that can handle either a list of SCHEMA_1_0 objects, or just a single SCHEMA_1_0 object.

File: kingpin/test/test_utils.py [hunk 2]
```diff
@@ -235,11 +235,19 @@ def test_repeating_log(self):
         logger = mock.Mock()  # used for tracking
 
         # Repeat this message 10 times per second
-        logid = utils.create_repeating_log(logger.info, 'test', seconds=0.1)
-        yield utils.tornado_sleep(0.45)  # Some process takes .4 <> .5 seconds
+        logid = utils.create_repeating_log(logger.info, 'test', seconds=0)
+        yield gen.moment
```
Comment (diranged) [inline kingpin/test/test_utils.py]: Not sure whats going on here -- explain it in person?

**Comment by diranged** [comment]
@siminm hey i like this change, do you think you can get it finished up this week and merged?

**Comment by siminm** [comment]
@matt added a couple of test, beefed up the IOLoop comment. Ready for review

**Comment by diranged** [comment]
it looks good -- but lets update our documentation to reflect this new use case..

**Comment by diranged** [comment]
👍  -- but please go update the docs with some examples/etc..

**Pull request merged**